projects/ngrx.io/content/guide/entity/recipes/additional-state-properties.md

```diff
@@ -0,0 +1,84 @@
+# Additional Entity State Properties Update
+
+For beginners it is very confusing how to update the additional state properties using adapter. So we are going to see how to update the Additional Entity state properties. In the below example we are going to use the [Entity Adapter](https://ngrx.io/guide/entity/adapter) example.
+
+Usage:
+
+we are declaring the `selectedUserId` as an additional entities state property.
+
+<code-example header="user.reducer.ts">
+import { EntityState, EntityAdapter, createEntityAdapter } from '@ngrx/entity';
+
+export interface User {
+  id: string;
+  name: string;
+}
+
+export interface State extends EntityState&lt;User&gt; {
+  // additional entities state properties
+  selectedUserId: number;
+}
+
+...
+
+export const adapter: EntityAdapter&lt;User&gt; = createEntityAdapter&lt;User&gt;({
+  ...
+});
+</code-example>
+
+Then create an action to update the `selectedUserId`
+
+<code-example header="user.actions.ts">
+import { createAction, props, union } from '@ngrx/store';
+import { Update } from '@ngrx/entity';
+
+import { User } from '../models/user.model';
+
+...
+
+export const selectUser = createAction('[User/API] Select User', props&lt;{ id: string }&gt;());
+
+...
+
+const all = union({
+  ...
+  selectUser,
+  ...
+});
+export type Union = typeof all;
+</code-example>
+
+Entity Adapter is only used to update the `EntityState` properties. Other than that additional state properties should be updated same as normal state properties like below.
+
+<code-example header="user.reducer.ts">
+import { EntityState, EntityAdapter, createEntityAdapter } from '@ngrx/entity';
+import { User } from '../models/user.model';
+import * as UserActions from '../actions/user.actions';
+
+export interface State extends EntityState&lt;User&gt; {
+  // additional entities state properties
+  selectedUserId: number | null;
+}
+
+export const adapter: EntityAdapter&lt;User&gt; = createEntityAdapter&lt;User&gt;();
+
+export const initialState: State = adapter.getInitialState({
+  // additional entity state properties
+  selectedUserId: null,
+});
+
+export function reducer(state = initialState, action: UserActions.Union): State {
+  switch (action.type) {
+    ...
+
+     case UserActions.selectUser.type: {
+      return { ...state, selectedUserId: action.id};

```

Comment:
I think we should also provide an example together with an entity helper function, e.g.

```ts
adapter.addMany(action.users, {...state, loaded: true});
```

Reply:
is this ok?
```javascript
on(UserActions.selectUser, state => {
   // updating without using entity helper function
    return { ...state, selectedUserId: action.id};
    OR
   // updating using entity helper function
    return adapter.addMany(action.users, { ...state, selectedUserId: action.id});
})
```

Reply:
I think your example was good, but we need one more example:
```
on(UserActions.selectUser, (state, { userId }) => {
    return { ...state, selectedUserId: userId };
}),
on(UserActions.loadUsers, (state, { users }) => {
    return adapter.addMany(users, { ...state, selectedUserId: null});
}),
```

Reply:
updated